File: app/src/main/java/com/v2/desafionubank/model/RequestLockCard.java
package com.v2.desafionubank.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by csanchez on 17/04/2018.
 */

public class RequestLockCard {
    private String comment;

    private List<RequestReasonDetails> reason_details = new ArrayList<>();

    public RequestLockCard(String comment, List<ReasonDetails> details) {
        this.comment = comment;

        for (ReasonDetails detail : details) {
            reason_details.add(new RequestReasonDetails(detail));
        }
    }
}

File: app/src/main/java/com/v2/desafionubank/model/RequestLockCard.kt
package com.v2.desafionubank.model

import java.util.*

/**
 * Created by CaioSChristino on 18/04/18.
 */

data class RequestLockCard(private val comment: String,
                           private val _details: List<ReasonDetails>) {
    private val reason_details = ArrayList<RequestReasonDetails>()

    init {
        for (detail in _details) {
            reason_details.add(RequestReasonDetails(detail))
        }
    }
}
